addons/hr_attendance/static/src/components/barcode_dialog/kiosk_barcode_dialog.js

```diff
@@ -0,0 +1,35 @@
+import { useState, onWillStart } from "@odoo/owl";
+import { BarcodeDialog } from "@web/webclient/barcode/barcode_scanner";
+import { useService } from "@web/core/utils/hooks";
+import { rpc } from "@web/core/network/rpc";
+export class AttendanceBarcodeDialog extends BarcodeDialog {
+    static template = "hr_attendance.BarcodeDialog";
+
+    /**
+     * @override
+     */
+    setup(){
+        super.setup();
+        this.orm = useService("orm");
+        this.state = useState({
+            barcode: false,
+        });
+        onWillStart( async () => {
+            this.isDemoState = await rpc('/hr_attendance/is_demo_state')
+        });
+    }
+
+    /**
+     * @override
+     */
+    get title() {
+        return "Scan your badge's barcode";
+    }
+
+    async setBadgeID(){
+        let barcode = this.state.barcode
+        if (barcode) {
+            await rpc('/hr_attendance/set_user_barcode', { barcode });
```

Comment:
check for security issue

Reply:
always use token for rpc calls from here :p